File: zig/src/template.zig
const std = @import("std");

pub const Input = struct {
    // TODO

    pub fn parse(allocator: std.mem.Allocator, input: []const u8) !Input {
        // TODO
    }

    pub fn deinit(self: Input) void {
        // TODO
    }
};

pub fn part1(allocator: std.mem.Allocator, s: []const u8) !u32 {
    const input = try Input.parse(allocator, s);
    defer input.deinit();
    // TODO
}

pub fn part2(allocator: std.mem.Allocator, s: []const u8) !u32 {
    const input = try Input.parse(allocator, s);
    defer input.deinit();
    // TODO
}

const TEST_INPUT = "";

test "part 1" {
    try std.testing.expectEqual(69, part1(std.testing.allocator, TEST_INPUT));
}

test "part 2" {
    try std.testing.expectEqual(1337, part2(std.testing.allocator, TEST_INPUT));
}

File: rust/src/template.rs
use std::{convert::Infallible, str::FromStr};

#[derive(Debug)]
struct Input {

}

impl FromStr for Input {
    type Err = Infallible;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        todo!()
    }
}

pub fn part1(input: &str) -> usize {
    todo!()
}

pub fn part2(input: &str) -> usize {
    todo!()
}

#[cfg(test)]
mod tests {
    use super::*;

    const INPUT: &str = "";

    #[test]
    fn test_part1() {
        assert_eq!(69, part1(INPUT));
    }

    #[test]
    fn test_part2() {
        assert_eq!(1337, part2(INPUT));
    }
}
